Database

jrans opened the pull request:
Implemented all necessary database functions for initial sprint:
- add user to db
- get user from db
- add users issues to db
- get users issues from db

fully tested and cleans up db after itself.

Best Wishes,

Jack, Daniel (and coke)

Danwhy committed:
creates redis pub and sub clients

Danwhy committed:
updated tests and wrote new tests for database functions

jrans committed:
restructured handler to only give whast necessary for the first sprint, i.e. adding all users issues and a user and getting them back

Danwhy committed:
100percent coverage on all db handlers

Danwhy committed:
Merge branch 'master' of github.com:dwyl/tudo into database

NataliaLKB commented on api/db_handlers.js:
```diff
@@ -1,47 +1,66 @@
 
+function getIssuesByUsername (client, username, callback) {
+    getIssueListByUsername(client, username, function(err, data) {
+        getIssuesByList (client, data, callback);
```
Can you please delete the space between the function and the brackets to keep it inline with how the rest of the functions are invoked?

jrans committed:
Update db_handlers.js

deleted a single space, 3 blank lines between functions and removed console log

jrans committed:
Update db_handlers.js

added 2 semicolons

jrans committed:
Update db_handlers.js

added back callback line

jrans committed:
Update db_handlers.js

added error to callback on line 30 for future use

jrans committed:
removed unnecessary callbacks for now
